pkg/controllers/awsloadbalancercontroller/controller.go

```diff
@@ -135,6 +138,20 @@ func (r *AWSLoadBalancerControllerReconciler) Reconcile(ctx context.Context, req
 		return ctrl.Result{}, fmt.Errorf("failed to ensure CredentialsRequest for AWSLoadBalancerController %q: %w", req.Name, err)
 	}
 
+	secretProvisioned, err := r.ensureCredentialsRequestSecret(ctx, credentialsRequest)
```

Comment:
So we are not ensuring the credentials secret here. We are just checking if it is available. So something like `provisioned, err := r.credentialsSecretProvisioned()` would be a better fit. Also if you get a `IsNotFound` error then just return false. It keeps the outer calling function cleaner.

Reply:
> Also if you get a IsNotFound error then just return false. It keeps the outer calling function cleaner.

It doesn't help much, since we still need to handle other types of errors (unless we suppress those errors) 

```go
if err == nil && !secretProvisioned {
       // update status and re-enqueue
} else if err != nil {
      // return non-NotFound errors
}
```

Reply:
All other errors we simply return an error even if `secretProvisioned` is true(because we never expect `secretProvisioned` to be true and for there to be an error). 

```
provisioned, err := r.credentialsSecretProvisioned()
if err!=nil {
   // this is only possible when there is some error other than not found
  // and we want to immediately retry because it's probably transient
  return err
}

r.updateStatus(..provisioned)
```